gwr/gwr.py

```diff
@@ -254,32 +255,22 @@ def fit(self, ini_params=None, tol=1.0e-5, max_iter=20, solve='iwls'):
             m = self.W.shape[0]
             params = np.zeros((m, self.k))
             predy = np.zeros((m, 1))
-            v = np.zeros((m, 1))
             w = np.zeros((m, 1))
-            z = np.zeros((m, self.n))
             S = np.zeros((m, self.n))
-            R = np.zeros((m, self.n))
             CCT = np.zeros((m, self.k))
-            #f = np.zeros((n, n))
-            p = np.zeros((m, 1))
             for i in range(m):
                 wi = self.W[i].reshape((-1,1))
-                rslt = iwls(self.y, self.X, self.family, self.offset, None,
-                ini_params, tol, max_iter, wi=wi)
+                rslt = iwls(self.y, self.X, self.family, self.offset, None, ini_params, tol, max_iter, wi=wi)
                 params[i,:] = rslt[0].T
                 predy[i] = rslt[1][i]
-                v[i] = rslt[2][i]
                 w[i] = rslt[3][i]
-                z[i] = rslt[4].flatten()
-                R[i] = np.dot(self.X[i], rslt[5])
-                ri = np.dot(self.X[i], rslt[5])
-                S[i] = ri*np.reshape(rslt[4].flatten(), (1,-1))
```

Comment:
`ri` is in shape of `(n, )`
`np.reshape(rslt[4].flatten(), (1,-1))` is in shape of `(1, n)`
This will be an element wise multiplication.